components/tf_cnn_benchmarks/reporter.py

```diff
@@ -0,0 +1,105 @@
+import csv
+import argparse
+import os
+import re
+import warnings
+
+
+"""
+Class LogParser
+
+Takes path to .txt file as the first argument
+and writes simplified log to .csv file
+
+
+Sample execution: python reporter.py path_to_log.txt path_to_table.csv
+
+Path to .csv file is optional. If not given,
+creates new file in the same folder as the script
+under the name log.csv
+
+"""
+
+
+class LogParser:
+    def __init__(self):
+        self.columns = [
+            'total_images/sec',
+            'TensorFlow',
+            'Model',
+            'Dataset',
+            'Mode',
+            'SingleSess',
+            'Batch_size',
+            'Num_batches',
+            'Num_epochs',
+            'Devices',
+            'Data_format',
+            'Layout_optimizer',
+            'Optimizer',
+            'Variables',
+            'Sync']
+        self.path_to_dataframe = self.parse_command_line().path_to_dataframe
+        self.path_to_file = self.parse_command_line().filename
+        self.log_dict = dict(zip(self.columns, [None] * len(self.columns)))
+        self.load_csv_log()
+
+    def parse_command_line(self):
+        self.parser = argparse.ArgumentParser(
+            description='Parser for filenames')
+        self.parser.add_argument('filename', type=str,
+                                 help='A required path to txt log file')
+        # Optional positional argument
+        self.parser.add_argument(
+            'path_to_dataframe',
+            type=str,
+            nargs='?',
+            default='log.csv',
+            help='An optional path to csv file')
+        return self.parser.parse_args()
+
+    def load_csv_log(self):
+        if not os.path.isfile(self.path_to_dataframe):
+            warnings.warn("Creating new csv log file")
```

Comment:
it is not necessary to have a warning of this, maybe change to info, or remove it